cli/azd/pkg/apphost/aca_ingress.go

```diff
@@ -0,0 +1,175 @@
+package apphost
+
+import (
+	"fmt"
+)
+
+// buildAcaIngress builds the Azure Container Apps ingress configuration from the provided bindings.
+func buildAcaIngress(bindings map[string]*Binding, defaultIngressPort int) (*genContainerAppIngress, error) {
+	if len(bindings) == 0 {
+		return nil, nil
+	}
+
+	if err := validateInput(bindings); err != nil {
+		return nil, err
+	}
+
+	endpointByContainerPort := make(map[string][]*Binding)
+	for _, binding := range bindings {
+		bindingKey := "default" // default is for those with no container port
```

Comment:
I don't know if it's acceptable to just use `0` here as an "automatically assigned port".

Reply:
It won't use 0, 
`http-only` endpoints can omit the TargetPort and they just get the `defaultPort` argument (either 80 or 8080)